hpx/lcos/local/receive_buffer.hpp

```diff
@@ -147,6 +147,13 @@ namespace hpx { namespace lcos { namespace local
             entry->set_value(std::move(val));
         }
 
+        // explicitly clear all sored data
+        void clear_all()
+        {
+            std::lock_guard<mutex_type> l(mtx_);
+            buffer_map_.clear();
```

Comment:
This clear shouldn't be needed. I think it should be an invariant of that class that the buffer_map is empty on destruction.